pkg/controller/llmisvc/workload_single_node.go

```diff
@@ -70,10 +75,17 @@ func (r *LLMInferenceServiceReconciler) expectedSingleNodeMainDeployment(ctx con
 		return nil, errors.New("llmSvc.Spec.Template must not be nil")
 	}
 
+	role := "decode"
+	if llmSvc.Spec.Prefill == nil {
+		role = "both"
+	}
+
 	labels := map[string]string{
 		"app.kubernetes.io/component": "llminferenceservice-workload",
 		"app.kubernetes.io/name":      llmSvc.GetName(),
 		"app.kubernetes.io/part-of":   "llminferenceservice",
+		"kserve.io/component":         "workload",
+		"llm-d.ai/role":               role,
```

Comment:
We have `singleNodeLabels`  func at the bottom - can't we re-use to reduce duplication?

Reply:
yes, I wanted to do it so bad but I didn't